### ISSUE TITLE
Loading of first definition after startup takes long time

### ISSUE THREAD

**Issue by DejanMilicic** [unisonweb/ucm-desktop#24]
After the app is started, first click on a definition in a code tree takes a significant amount of time.
This "cold start" takes so long that it made me think, on several occasions, that application is not working at all.

**Comment by hojberg** [comment]
Yeah I'm not sure how to best deal with that. We could just try and preload a random definition or put in a loader.

@ChrisPenner any thoughts?

**Comment by ChrisPenner** [comment]
@hojberg which APIs are being called as part of startup? Do we actually need to load the root branch to start loading definitions?

**Comment by hojberg** [comment]
@ChrisPenner the browse api for the sidebar is the main one I think. Its fast enough. It's the first definition fetch that ends up being slow. Not sure what the difference is since the browse api does list definititions.

**Comment by ChrisPenner** [comment]
Ah, yeah I suppose in order to pretty-print we need to load the whole branch :'(

I'm surprised it's feeling so slow that it feels completely unresponsive though; for most projects I'm finding UCM is responsive within a second or two, but perhaps this is a particularly large project, or on a slower laptop than mine.

Unfortunately I think the only short-term thing we can do is add a loading bar or something, I'd still like to move more of the pretty-printing to database indexes which would open up more solutions, but that's a significant effort to take on.
